### PR TITLE
feat: trade component revision

### PR DESCRIPTION
**Pull request by dcocking**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue
https://cybrid.atlassian.net/browse/CYB-75
https://cybrid.atlassian.net/browse/CYB-370

### Why
Revision and continuation of the trade component.

### How
- Expanded the host config to accept a customer GUID 
- Some refactoring of `trade.component.ts` to align with the `postQuoteBankModel`
- Added asset select icons
- Fixes for cypress `tsconfig.ts` collisions

### CODE REVIEW

File: library/src/app/components/trade/trade.component.ts [hunk 1]
```diff
@@ -70,17 +59,23 @@ export class TradeComponent implements OnInit, OnDestroy {
...
   });
 
-  quote: Quote = {
-    asset: Constants.ASSET,
-    counterAsset: Constants.COUNTER_ASSET,
+  postQuoteBankModel: PostQuoteBankModel = {
+    customer_guid: '',
+    symbol: '',
+    side: PostQuoteBankModel.SideEnum.Buy
+  };
```
Comment (brent-cybrid) [inline library/src/app/components/trade/trade.component.ts:67]: Is this the model used to communicate with the REST API?

File: src/shared/constants/constants.ts [hunk 2]
```diff
@@ -20,18 +20,62 @@ export class Constants {
   static MIN_FRACTION_DIGITS = 2;
   static THEME = THEME.LIGHT;
   static ICON_HOST = 'https://images.cybrid.xyz/color/';
-  static COUNTER_ASSET: AssetBankModel = {
+  static CAD_ICON =
+    'https://upload.wikimedia.org/wikipedia/commons/c/cf/Flag_of_Canada.svg'; // Temporary Wikipedia svg for demo purposes
+  static USD_ICON =
+    'https://upload.wikimedia.org/wikipedia/en/a/a4/Flag_of_the_United_States.svg'; // Temporary Wikipedia svg for demo purposes
```
Comment (brent-cybrid) [inline src/shared/constants/constants.ts:26]: We should put these in our CDN

File: library/src/app/components/trade/trade.component.ts [hunk 3]
```diff
@@ -208,6 +234,52 @@ export class TradeComponent implements OnInit, OnDestroy {
...
+          delete this.postQuoteBankModel.deliver_amount;
+          if (this.postQuoteBankModel.side == 'buy') {
+            this.postQuoteBankModel.receive_amount = this.assetPipe.transform(
+              amount,
+              this.asset,
+              'base'
+            ) as number;
+          } else {
+            this.postQuoteBankModel.receive_amount =
+              amount * this.price.sell_price!;
```
[the remark this answers is not in the document]
Comment (dcocking) [reply library/src/app/components/trade/trade.component.ts:251]: No it's the fraction of whole coins.

if `input = 'receive_amount'` and  `side = 'sell'` and the input field value is 2 BTC it looks like this:

2BTC * 20600 Satoshi

It seems like a conflict of units but the price is always returned as the base unit value of 1 whole trade unit.

File: library/src/app/components/trade/trade.component.ts (hunk 1 above)
Comment (dcocking) [reply library/src/app/components/trade/trade.component.ts:67]: Yeah, minus the `receive_amount` or `deliver_amount`.

File: src/shared/services/quote/quote.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    postQuote = service.getQuote(
+      amount,
+      input,
+      PostQuoteBankModel.SideEnum.Sell,
+      asset,
+      counter_asset
+    );
+    expect(postQuote.receive_amount).toEqual(1000); // Cents
+  });
+});
```
Comment (brent-cybrid) [inline src/shared/services/quote/quote.service.spec.ts:111]: What about tests for very large numbers?

File: library/src/app/components/trade/trade.component.ts [hunk 5]
```diff
@@ -114,74 +108,93 @@ export class TradeComponent implements OnInit, OnDestroy {
...
-  getRouterParams(): void {
+  /*
+  Set the current asset and counter-asset from routing data.
+  Set the list of available crypto and fiat assets.
+  * */
+  getAssets(): void {
     this.route.queryParams
       .pipe(
-        takeUntil(this.unsubscribe$),
+        take(1),
```
[the remark this answers is not in the document]
Comment (dcocking) [reply library/src/app/components/trade/trade.component.ts:118]: Yes I only need the router params when the component initializes. It’s really the same thing but this way it auto unsubscribes itself.

File: src/shared/services/quote/quote.service.spec.ts (hunk 4 above)
Comment (brent-cybrid) [reply src/shared/services/quote/quote.service.spec.ts:114]: Looks good. Can we add a test for the largest and smallest amount of ETH?

**Review by brent-cybrid — APPROVED** [review]

File: src/shared/services/quote/quote.service.spec.ts (hunk 4 above)
Comment (dcocking) [reply src/shared/services/quote/quote.service.spec.ts:148]: Done. Caveat here is that until we convert the openApi generator `PostQuoteBankModel` type to use `BigInt` then we are limited by the amount of ETH we can handle.

**Pull request merged**